crates/client/src/https_client_connection.rs

```diff
@@ -38,14 +37,28 @@ impl<T> HttpsClientConnection<T> {
     ///
     /// * `name_server` - address of the name server to use for queries
     #[allow(clippy::new_ret_no_self)]
-    pub fn new() -> HttpsClientConnectionBuilder {
-        HttpsClientConnectionBuilder::new()
+    pub fn new() -> HttpsClientConnectionBuilder<T> {
+        HttpsClientConnectionBuilder::default()
+    }
+}
+
+impl<T: RuntimeProvider> HttpsClientConnection<T> {
+    /// Creates a new client connection with runtime provider.
+    ///
+    /// *Note* this has side affects of binding the socket to 0.0.0.0 and starting the listening
+    ///        event_loop. Expect this to change in the future.
+    ///
+    /// # Arguments
+    ///
+    /// * `runtime` - runtime provider to be used for establishing an HTTPS connection.
+    pub fn with_runtime(runtime: T) -> HttpsClientConnectionBuilder<T> {
+        HttpsClientConnectionBuilder::new(runtime)
```

Comment:
Ok, this is good. I like cleaning it up so that we don't have the Default requirement on Runtime, which might be awkward.